nx/test/nx/backend_test.exs

```diff
@@ -0,0 +1,103 @@
+defmodule Nx.BackendTest do
```

Comment:
Suggestion to rename this to `nx/optional_test.exs`. Especially because we will need to add more tests once we add the `defn` implementation?